pymedphys/_gui/streamlit/pseudonymise.py

```diff
@@ -0,0 +1,284 @@
+"""Streamlit Gui for Pseudonymise
+"""
+# Copyright (C) 2020 Stuart Swerdloff
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import base64
+import datetime
+import io
+import pathlib
+from typing import List
+from zipfile import ZIP_DEFLATED, ZipFile
+
+import streamlit as st
+
+import pydicom
+
+from pymedphys._dicom.anonymise import anonymise_dataset
+from pymedphys._dicom.constants.core import DICOM_SOP_CLASS_NAMES_MODE_PREFIXES
+from pymedphys._dicom.utilities import remove_file
+from pymedphys.experimental import pseudonymisation as pseudonymisation_api
+
+# We create a downloads directory within the streamlit static asset directory
+# and we write output files to it, similar to approach in streamlit forum
+DOWNLOADS_PATH = pathlib.Path(st.__path__[0]).joinpath("static", "downloads")
+DOWNLOADS_PATH.mkdir(parents=True, exist_ok=True)
```

Comment:
I believe this here has actually combined two approaches. When going the b64 encoding route, don't actually need to store the data into Streamlit's static directory. In fact, don't actually need to store the data to disk at all.

Instead, can just open an in memory file object using `io.BytesIO`:

```python
in_memory_temp_file = io.BytesIO()
pydicom.dcmwrite(in_memory_temp_file, ds_input)
myzip.write(in_memory_temp_file, anon_filename)
```